Sources/Networking/Modifiers/Interceptors/Authorization/AuthorizationManaging.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  AuthorizationManaging.swift
+//  
+//
+//  Created by Dominika Gajdová on 20.12.2022.
+//
+
+import Foundation
+
+/// AuthorizationManaging authorizes requests and manages refresh token mechanism
+/// AuthorizationStorageManaging is required to read & store `AuthorizationData`
+public protocol AuthorizationManaging {
+    var storage: any AuthorizationStorageManaging { get }
```

Comment:
Should we have a direct access to storage in the protocol?

Reply:
Well we want for AuthorizationManagers to define this property.